build-system/server/lazy-build.js

```diff
@@ -63,8 +68,16 @@ async function lazyBuild(url, matcher, bundles, buildFunc, next) {
   if (match && match.length == 2) {
     const name = maybeGetUnminifiedName(bundles, match[1]);
     const bundle = bundles[name];
-    if (bundle) {
+
+    if (bundle && argv.compiled) {
       await build(bundles, name, buildFunc);
+    } else if (bundle && !esbuildWatchedExtensions.has(name)) {
+      // For unminified builds, only initiate the build/watcher.
+      esbuildWatchedExtensions.add(name);
+      await buildFunc(bundles, name, {
+        minify: false,
+        watch: true,
+      });
```

Comment:
Two things that may not be actual problems:
1. The difference between calling `buildFunc` directly and having it called via `build()` is that `build()` also checks for a pending build before triggering a new build. (This used to be important when builds took a few seconds and we had races.)
2. From reading this logic, it looks like we do not call `buildFunc` when `esbuildWatchedExtensions.has(name) == true`. I assume this is intentional since `build()` was bailing when `bundle.watched == true`, right?

Reply:
For (1), is the goal that we wait for the bundle to be updated before we serve the file? The usefulness is definitely lowered when the build completes this quickly (< 100ms), but isn't gone. I'm unsure how to add that back right now since we only have a callback when the rebuild completes instead of when it begins 🤔.

For (2), the reason I don't call buildFunc is because all of the state management / filewatching etc. is handled by `esbuild`. We don't need  to manage anything else once it starts.

---

It is totally possible we'll want to switch back to manually triggering rebuilds at some point, especially if we don't get a clean resolution to (1)

Reply:
Aha, I see. I added a second round of comments suggesting this logic be moved out of `lazyBuild()` and into `build()`. Maybe that will help resolve (1).